app/src/main/java/io/intrepid/contest/screens/sendinvitations/SendInvitationsPresenter.java

```diff
@@ -122,17 +122,27 @@ public void onOptionsItemSelected(int itemId) {
                 break;
             case R.id.send_invitations_skip_menu_action:
                 view.showMessage("TODO: Go to next screen");
+            case android.R.id.home:
+                onBackButtonClicked();
         }
     }
 
     private void displayMenuItems() {
         if (displayMenuItems) {
             view.setActionBarTitle(R.string.invite_contestants_bar_title);
+            view.setActionBarDisplayHomeAsUpEnabled(false);
             view.showSendInvitationsMenuItem(!selectedContactList.isEmpty());
             view.showSendInvitationsSkipMenuItem(selectedContactList.isEmpty());
         } else {
             view.showSendInvitationsMenuItem(false);
             view.showSendInvitationsSkipMenuItem(false);
         }
     }
+
+    @Override
+    public void onBackButtonClicked() {
+        if (contactSelectionEnabled) {
```

Comment:
This conditional setup is very straightforward. A bit out of the scope of this review, but I think the displayMenuItems() chunk can also be simplified in this way. The conditional check in the displayMenuItems  can go in your onCreateOptionsMenu. And the displayMenuItem() method should just be responsible for displaying the items. Your onCreateOptionsMenu code would look like```if (displayMenuItems) displayMenuItems(); The other alternative would be to pass in a boolean to the displayMenuItems() method